src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -58,7 +57,7 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(string id,
                     {
                         new LibraryDependency
                         {
-                            LibraryRange = new LibraryRange(id, new VersionRange(NuGetVersion.Parse(version)), LibraryDependencyTarget.Package),
+                            LibraryRange = new LibraryRange(libraryIdentity.Name, new VersionRange(libraryIdentity.Version), LibraryDependencyTarget.Package),
```

Comment:
This could potentially resolve to versions different than the specified because it's the equivalent of: 

`[version, )`, you might want `[version]`

Or is this on purpose? 
Later in the result you are looking for all the exact versions anyways.

Reply:
No you're right, we want exact versions only.  I'll update

Reply:
@nkolev92 please confirm that I did this right to ensure that the version is exact:

```C#
LibraryRange = new LibraryRange(
    libraryIdentity.Name,
    new VersionRange(
        minVersion: libraryIdentity.Version,
        includeMinVersion: true,
        maxVersion: libraryIdentity.Version,
        includeMaxVersion: true),
    LibraryDependencyTarget.Package),
```

Reply:
Yup, that looks right!